internal/commands/list_buildpack_registries_test.go

```diff
@@ -0,0 +1,94 @@
+package commands_test
+
+import (
+	"bytes"
+	"testing"
+
+	h "github.com/buildpacks/pack/testhelpers"
+
+	"github.com/buildpacks/pack/internal/commands"
+
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack/internal/config"
+	ilogging "github.com/buildpacks/pack/internal/logging"
+	"github.com/buildpacks/pack/logging"
+)
+
+func TestListBuildpacksRegistries(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+
+	spec.Run(t, "Commands", testListBuildpackRegistriesCommand, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testListBuildpackRegistriesCommand(t *testing.T, when spec.G, it spec.S) {
+	var (
+		command *cobra.Command
+		logger  logging.Logger
+		outBuf  bytes.Buffer
+		cfg     config.Config
+	)
+
+	it.Before(func() {
+		logger = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+		cfg = config.Config{
+			DefaultRegistryName: "private registry",
+			Registries: []config.Registry{
+				{
+					Name: "public registry",
+					Type: "github",
+					URL:  "https://github.com/buildpacks/public-registry",
+				},
+				{
+					Name: "private registry",
+					Type: "github",
+					URL:  "https://github.com/buildpacks/private-registry",
+				},
+				{
+					Name: "personal registry",
+					Type: "github",
+					URL:  "https://github.com/buildpacks/personal-registry",
+				},
+			},
+		}
+		command = commands.ListBuildpackRegistries(logger, cfg)
+	})
+
+	when("#ListBuildpackRegistries", func() {
+		it("should list all registries", func() {
+			h.AssertNil(t, command.Execute())
+
+			h.AssertContains(t, outBuf.String(), "public registry")
+			h.AssertContains(t, outBuf.String(), "* private registry")
+			h.AssertContains(t, outBuf.String(), "personal registry")
+		})
+
+		it("should list registries in verbose mode", func() {
+			logger := ilogging.NewLogWithWriters(&outBuf, &outBuf, ilogging.WithVerbose())
+			command = commands.ListBuildpackRegistries(logger, cfg)
+
+			h.AssertNil(t, command.Execute())
+
+			h.AssertContains(t, outBuf.String(), "public registry")
+			h.AssertContains(t, outBuf.String(), "https://github.com/buildpacks/public-registry")
+
+			h.AssertContains(t, outBuf.String(), "private registry")
+			h.AssertContains(t, outBuf.String(), "https://github.com/buildpacks/private-registry")
+
+			h.AssertContains(t, outBuf.String(), "personal registry")
+			h.AssertContains(t, outBuf.String(), "https://github.com/buildpacks/personal-registry")
+		})
+
+		it("should list nothing when NO registries exist", func() {
```

Comment:
Is there not the idea of a built-in default registry? Aka "official"? In my mind, I had envisioned it as being set automatically (out-of-the-box). It was up in the air as to whether it could be removed or not. If it can't be removed then would we ever have this case?

Also, I agree with @dfreilich that we should "Hint" as to how to add a registry when listing the registries and there are no custom registries set.

Reply:
@jromero I can add hint.  

With regards to always having a "`official`" registry, I would have to double check or maybe @jkutner knows off the top of his head.  However, we don't have complete control over this file, so if the user wanted to they could intentionally (or accidentally) delete it.  If this were the case what's the expected behavior?  Should we automatically re-create the `official` registry and always display it at a minimum?

Reply:
> With regards to always having a "official" registry, I would have to double check or maybe @jkutner knows off the top of his head. However, we don't have complete control over this file, so if the user wanted to they could intentionally (or accidentally) delete it. If this were the case what's the expected behavior? Should we automatically re-create the official registry and always display it at a minimum?

It may be worth discussing in a separate issue as to not overload this one. I believe a strategy that _may_ work is to hardcode it in memory instead of appending it to the config file. A similar example of this would be trusted-builders in which case we list trusted-builders as "user-defined trusted builder" (from config.toml) + "suggested builders" (from hard coded values).

Reply:
Created https://github.com/buildpacks/pack/issues/842 to continue conversation there.

Reply:
I commented on the issue, but this test should check that the _default_ registry is listed when there are no entries in the config file. That default registry (the official registry) should be hard-coded into pack. We can allow it to be overridden in the config though.